### PR TITLE
Sync-related consistency improvements

### PR DESCRIPTION
**Pull request by rsdy**
Fixes bugs where:
 * Upstream repository changes aren't consistently picked up by the indexer
 * Race condition where Bloop tries to update repositories before credentials are available
 * If more than the number of worker threads repositories are queued, they aren't immediately reflected on the UI, only when they become active in the sync queue.

### CODE REVIEW

File: server/bleep/src/background/sync.rs [hunk 1]
```diff
@@ -87,23 +87,46 @@ impl Drop for SyncHandle {
...
-            reporef,
             new_branch_filters,
             exited,
             exit_signal,
-        }
-        .into()
+        };
+
+        _ = sh
+            .create_new(|| {
```
Comment (ggordonhall) [inline server/bleep/src/background/sync.rs:108]: Can we just roll `create_new` into this method? Would make this logic easier to follow

File: server/bleep/src/periodic/remotes.rs [hunk 2]
```diff
@@ -138,6 +138,10 @@ async fn update_credentials(app: &Application) {
 }
 
 pub(crate) async fn check_repo_updates(app: Application) {
+    while app.credentials.github().is_none() {
```
Comment (ggordonhall) [inline server/bleep/src/periodic/remotes.rs:141]: Could we get stuck in this state?

File: server/bleep/src/repo/iterator/git.rs [hunk 3]
```diff
@@ -58,7 +58,7 @@ impl GitWalker {
         let head_name = head
             .clone()
             .try_into_referent()
-            .map(|r| r.name().to_owned());
+            .map(|r| format!("origin/{}", human_readable_branch_name(&r)));
```
Comment (ggordonhall) [inline server/bleep/src/repo/iterator/git.rs:61]: Can you add a comment explaining why we do this?

File: server/bleep/src/periodic/remotes.rs (hunk 2 above)
Comment (rsdy) [reply server/bleep/src/periodic/remotes.rs:141]: Yes, if there's no authentication we'll never enter this loop.

**Review by ggordonhall — APPROVED** [review]

**Pull request merged**